Expensify/App
# Add Validation to BankAccount + CompanyStep + better handling for Plaid

cc @roryabraham 

### Details
Adds validation to forms in the  `BankAccount` / `CompanyStep`

### Fixed Issues
Fixes https://github.com/Expensify/Expensify/issues/167761

### Tests
1. Try to break the manual account form + company information step in various ways
2. Verify helpful error messages appear in the form of growls.

![2021-06-18_14-12-55](https://user-images.githubusercontent.com/32969087/122626584-1ba53780-d047-11eb-848b-52cbe85d3f85.png)
![2021-06-18_14-11-12](https://user-images.githubusercontent.com/32969087/122626587-1d6efb00-d047-11eb-8995-78af96d81af5.png)
![2021-06-18_14-11-06](https://user-images.githubusercontent.com/32969087/122626589-1e079180-d047-11eb-8b22-b23ba73248d7.png)
![2021-06-18_14-10-56](https://user-images.githubusercontent.com/32969087/122626590-1f38be80-d047-11eb-988b-e9235899dc67.png)
![2021-06-18_14-10-50](https://user-images.githubusercontent.com/32969087/122626592-1f38be80-d047-11eb-94b3-36df0c395e6f.png)

### QA Steps
No QA

### Tested On


### Screenshots


#### Web


#### Mobile Web


#### Desktop


#### iOS


#### Android
